inc/pendingreason.class.php

```diff
@@ -0,0 +1,275 @@
+<?php
+/**
+ * ---------------------------------------------------------------------
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2015-2021 Teclib' and contributors.
+ *
+ * http://glpi-project.org
+ *
+ * based on GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2003-2014 by the INDEPNET Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+if (!defined('GLPI_ROOT')) {
+   die("Sorry. You can't access this file directly");
+}
+
+/**
+ * ITILCategory class
+**/
+class PendingReason extends CommonDropdown
+{
+   // From CommonDBTM
+   public $dohistory = true;
+
+   // From CommonDBTM
+   public $can_be_translated = true;
+
+   // Rights managment
+   public static $rightname = 'pendingreason';
+
+   public static function getTypeName($nb = 0) {
+      return _n('Pending reason', 'Pending reasons', $nb);
+   }
+
+   public function getAdditionalFields() {
+      return [
+         [
+            'name'  => 'auto_bump',
+            'label' => __('Automatic follow-up'),
+            'type'  => '',
+            'list'  => true
+         ],
+         [
+            'name'      => 'itilfollowuptemplates_id',
+            'label'     => __('Followup template used for automatic follow-up'),
+            'type'      => 'dropdownValue',
+            'list'      => true
+         ],
+         [
+            'name'  => 'auto_solve',
+            'label' => __('Automatic resolution'),
```

Comment:
The name/label suggests that it is a boolean, and not a maximum count.

Reply:
I'm OK to change auto_solve and auto_resolved fields name as you suggested earlier but I'm worried about the users already using this feature (e.g. the internal ref).
If we do this, would it be ok to keep a small code in the migration to rename the column for the users who already have these field ?

Reply:
Indeed, you could add a migration for both fields (and add a comment in migration code to indicate that fields were renamed). We already done that in the past.

Reply:
Ok, I'll work into it then :)
`auto_bump` -> `bump_frequency` as you suggested earlier seems good to me (or maybe it should it be `followup_frequency` now ?)
`auto_solve` -> `bumps_before_resolution` / `followups_before_resolution` ?

Reply:
OK for:
`followup_frequency`
`followups_before_resolution`